src/datachain/lib/dc/storage_pattern.py

```diff
@@ -0,0 +1,299 @@
+import glob
+from typing import TYPE_CHECKING, Union
+
+from datachain.lib.listing import ls
+
+if TYPE_CHECKING:
+    from .datachain import DataChain
+
+
+def validate_cloud_bucket_name(uri: str) -> None:
+    """
+    Validate that cloud storage bucket names don't contain glob patterns.
+
+    Args:
+        uri: URI to validate
+
+    Raises:
+        ValueError: If a cloud storage bucket name contains glob patterns
+    """
+    if not any(uri.startswith(scheme) for scheme in ["s3://", "gs://", "az://"]):
```

Comment:
No `hf`? Do we really want to hard code the list? Is there a better way to handle this?

Reply:
good catch. 

1. I added hf with test.

2. I was not able to find a single check for this in fsspec/client - I extracted this to `fsspec.py`